### PR TITLE
[5.6] Fix ordering of migrations in status command

### PR DESCRIPTION
**Pull request by tomgrohl**
First pull request so bear with me.

This relates to issue #24107 that I opened.

This PR uses the ran migrations and diffs with all the migration files to get any unran ones.

The resulting array thats passed to the table is ordered correctly by name and batch

### CODE REVIEW

File: src/Illuminate/Database/Console/Migrations/StatusCommand.php [hunk 1]
```diff
@@ -75,14 +75,23 @@ public function handle()
...
+            $migrations[] = ['<info>Y</info>', $migration, $batch];
+        }
+
+        // Diff to find ones that aren't ran
+        $diff = Collection::make($this->getAllMigrationFiles())
+            ->map(function ($migration) {
+                return $this->migrator->getMigrationName($migration);
+            })->diff($ran);
+
+        foreach ($diff as $migrationName) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/Migrations/StatusCommand.php]: Why the foreach, and not just another map?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by GrahamCampbell** [comment]
Thanks for the PR. I've left a review comment. :)

File: src/Illuminate/Database/Console/Migrations/StatusCommand.php (hunk 1 above)
Comment (tomgrohl) [reply src/Illuminate/Database/Console/Migrations/StatusCommand.php]: True. I'll replace the foreach's with collections

**Comment by tomgrohl** [comment]
I've updated it to use collections instead of foreach's. Not sure if thats any better

**Comment by GrahamCampbell** [comment]
Could you add a test that failed before, but passes after please. Then I'd be more confident this works. :)

**Comment by tomgrohl** [comment]
Sure leave it with me.

**Comment by tomgrohl** [comment]
Added a test for the change, test did fail originally because I had the parameters the wrong way round on the first map call. This is now fixed.

I've used Symfony's CommandTester so I could add an assertion for the table output from the command. I ran the test with the original`getStatusFor` method, which fails and then afterwards with my change, which passes.

**Review by GrahamCampbell — APPROVED** [review]

**Comment by GrahamCampbell** [comment]
Great, this looks solid. 👍

**Comment by taylorotwell** [comment]
Broken on my machine:

image

**Comment by tomgrohl** [comment]
Can I assume because his has been closed you don't want me to investigate and fix?

**Comment by tomgrohl** [comment]
@X-Coder264 Ah thanks. I'll do that.

**Comment by tomgrohl** [comment]
@taylorotwell any chance you can run the command with `-v` so I can try and debug the issue please? I'm having trouble replicating it.

**Comment by tomgrohl** [comment]
I'm having trouble replicating the error @taylorotwell got. I've tried it with different migration states, unran + no ran, unran, ran, no migrations with multiple batches.

Usually and `Array to string conversion` error with an `array_diff` happens if an array is multi-dimensional, but as far as I can see there not.

**Pull request closed without merging**